### ISSUE TITLE
ZeroDivisionError when trying to train Electra

### ISSUE THREAD

**Issue by SpiderBraids** [ThilinaRajapakse/simpletransformers#557]
**Describe the bug**
I have been trying to train the Electra NLP model from scratch on my data, but while it trains properly to completion with smaller datasets, when I try to train on larger datasets, the running loss will randomly become NaN, leading to loss scaler 0 repeatedly reducing loss scale by 2, eventually reaching 0 and producing a ZeroDivisionError. I use the same dataset to generate the tokenizer vocab files, and I have verified that all data in both training and testing datasets can be completely tokenized without producing a NaN input.

**Expected behavior**
I expect training to complete without errors, regardless of number of epochs and training dataset size, but it seems that I randomly get these errors, especially when trying to train with large datasets.

**Screenshots**
Pictures of the relevant Docker logs:
- NaN running loss and loss scale being reduced by 2
![image](https://user-images.githubusercontent.com/54842128/86893868-77272f80-c12c-11ea-8a23-fa0f48c13e4c.png)
- Loss scale reaching 0 and beginning of error trace
![image](https://user-images.githubusercontent.com/54842128/86893904-8312f180-c12c-11ea-9dc5-e87385734a80.png)
- End of error trace
![image](https://user-images.githubusercontent.com/54842128/86893928-8ad29600-c12c-11ea-864d-34ada786387b.png)

**Desktop**
I run my training in a Docker on an Ubuntu 4.15.0 box (64-bit).

**Additional context**
I am currently using a Tokenizer class trivially inherited from the standard ElectraTokenizer, but I have also had this happen using the standard "[Training an ELECTRA model from scratch](https://simpletransformers.ai/docs/lm-minimal-start/#training-an-electra-model-from-scratch)" code with the standard ElectraTokenizer.

**Comment by ThilinaRajapakse** [comment]
Can you try training without `fp16`?
